### DISCUSSION TITLE
RSBuild with Module federation issue while have multiple child apps

### DISCUSSION THREAD

**Discussion by ambujbpl** [web-infra-dev/rsbuild#2906] [Q&A]
Hello RSBuild Team, 
I am trying to set two child apps with host app.
and my second app not loaded fine with host app but when I try to run and configure single child app than both apps working fine (individually).

### **host rsbuild.config.js**
 ```
moduleFederation: {
        options: {
            name: 'host',
            remotes: {
                app1: 'app1@http://localhost:5000/remoteEntry.js',
                app2: 'app2@http://localhost:4000/remoteEntry.js',
            },
            shared: {
                ...dependencies,
                react: {
                    singleton: true,
                    requiredVersion: dependencies.react,
                },
                'react-dom': {
                    singleton: true,
                    requiredVersion: dependencies['react-dom'],
                },
            },
        },
```

### **app 1 rsbuild.config.js**
```
 moduleFederation: {
    options: {
        name: 'app1',
        exposes: {
            './App': './src/App',
        },
        filename: 'remoteEntry.js',
        shared: {
            ...dependencies,
            react: {
                singleton: true,
                requiredVersion: dependencies.react,
            },
            'react-dom': {
                singleton: true,
                requiredVersion: dependencies['react-dom'],
            },
        },
    },
}
``` 
### **app 2 rsbuild.config.js**
```
 moduleFederation: {
    options: {
        name: 'app2',
        exposes: {
            './App': './src/App',
        },
        filename: 'remoteEntry.js',
        shared: {
            ...dependencies,
            react: {
                singleton: true,
                requiredVersion: dependencies.react,
            },
            'react-dom': {
                singleton: true,
                requiredVersion: dependencies['react-dom'],
            },
        },
    },
}
```
I am not able to get one child app helper minified js files (screenshot attached)
<img width="414" alt="Image (1)" src="https://github.com/user-attachments/assets/002353d2-4319-4026-b194-2fa866211106">

**Comment by chenjiahan** [comment]
Hi, can you provide a reproduction for this issue?

**Comment by capndave** [comment]
I work on a team with @ambujbpl. The problem was that both of the child applications were named "shared" in their `package.json` files. When the `remoteEntry.js` file for them was created, they provided conflicting information to the parent (module-federated) app. See [here](https://github.com/webpack/webpack/issues/15582#issuecomment-1140638060) for more info.

If anyone sees similar weirdness, especially if they encounter a `Cannot read properties of undefined (reading 'call')` error, make sure the name values in each `package.json` are unique!
